source/common/http/conn_manager_utility.cc

```diff
@@ -107,8 +110,17 @@ Network::Address::InstanceConstSharedPtr ConnectionManagerUtility::mutateRequest
         Utility::appendXff(request_headers, *connection.remoteAddress());
       }
     }
-    request_headers.insertForwardedProto().value().setReference(
-        connection.ssl() ? Headers::get().SchemeValues.Https : Headers::get().SchemeValues.Http);
+    if (trusted_forwarded_proto) {
+      // Set x-forwarded-proto unless it already exists and was forwarded on by a trusted proxy.
+      if (xff_num_trusted_hops == 0 || request_headers.ForwardedProto() == nullptr) {
```

Comment:
Should this respect `skip_xff_append`?

Reply:
I don't _think_ so.  
skip_xff_append is to make the proxy invisible from a fowarding perspective.  We don't have a general "stop appending headers" config, we do them one at a time (there's already config knobs to avoid adding Via and I'm adding one for Server shortly).  Meanwhile if we skip adding it at all we'd break things in the router, so all this code does is decide if we set an original value or trust the downstream value, which I think is orthogonal from if we want this proxy to be a visible hop in xff at the usptream server.

Reply:
My thinking was that `X-Forwarded-For` and `X-Forwarded-Proto` are kind of coupled together (and they are part of the same "entry" in the [Forwarded (RFC7239)](https://tools.ietf.org/html/rfc7239) header), so it would be weird seeing `X-Forwarded-Proto` alone...

But if you think that's fine, then it's fine. I just wanted to make sure you didn't miss this. The code is definitely an improvement over what we have now. Thanks!